src/qilib/configuration_helper/adapters/d5a_instrument_adapter.py

```diff
@@ -40,9 +43,9 @@ class D5aInstrumentAdapter(ReadOnlyConfigurationInstrumentAdapter, SpiModuleInst
 
     def __init__(self, address: str, instrument_name: Optional[str] = None) -> None:
         super().__init__(address, instrument_name)
-        self._instrument = D5a(self._instrument_name, self._spi_rack, self._module_number, mV=MV,
-                               inter_delay=INTER_DELAY,
-                               reset_voltages=RESET_VOLTAGE, dac_step=DAC_STEP)
+        self._instrument: Instrument = D5a(self._instrument_name, self._spi_rack, self._module_number, mV=MV,
```

Comment:
self._instrument is typed as optional in the  base class. This leads to all kinds of type hinting errors because in theory the self._instrument can be None.
In the adapter classes the self._instrument is explicitly set in the constructor. But even then, usage of _instrument in these adapter classes (without a check for None) results in an mypy error. To prevent that, we 'casted' the _instrument of the bases class to the instrument type itself, not Optional, so no errors.

While type hinting I defined self._instrument not optional anymore. So I could type self._instrument as Instrument again. That's what you see here.

BUT: I decided to revert that change, so instrument is Optional again. Therefore for the reason I described earlier, all assignments to _instrument are explicitly 'casted' to the instrument again. This to make it clear what types we are talking about.

BTW: the spi_rack_instrument_adapter and spi_module_instrument_adapter assigned a completely different class (spi_rack) to _instrument which of course is wrong. Together with the review comments I rearranged some things there. Now these 2 classes don't use the member _instrument of the base class anymore but a new member _spi_rack. I overrid a property and method to handle this change correctly. Please look at that part in the re-review also.